File: winner_linkedlist/winner.js
function ListNode(val, next) {
	this.val = val === undefined ? 0 : val;
	this.next = next === undefined ? null : next;
}
/**
 * @param {ListNode} head
 * @returns {string}
 */
var gameResult = function (head) {
	let [evenScore, oddScore] = [0, 0];

	while (head !== null && head.next !== null) {
		let [even, odd] = [head.val, head.next.val];
		head = head.next.next;

		if (even > odd) {
			evenScore++;
		} else if (odd > even) {
			oddScore++;
		}
	}

	if (evenScore === oddScore) {
		return "Tie";
	}

	return evenScore > oddScore ? "Even" : "Odd";
};

File: winner_linkedlist/winner.go
package winnerlinkedlist

type ListNode struct {
	Val  int
	Next *ListNode
}

func gameResult(head *ListNode) string {
	oddScore, evenScore := 0, 0

	for head != nil && head.Next != nil {
		even, odd := head.Val, head.Next.Val

		if even > odd {
			evenScore++
		} else if odd > even {
			oddScore++
		}
		head = head.Next.Next
	}

	return findMax(oddScore, evenScore)
}

func findMax(odd, even int) string {
	if odd > even {
		return "Odd"
	} else if even > odd {
		return "Even"
	}

	return "Tie"
}
